### PR TITLE
Side Bars

### PR DESCRIPTION
**Pull request by adamgordonbell**
This is the change to add related items and earthly tie ins to the side bars of the blog posts. Test on makefile posts shows this very likely increases time spent on site. 

Related items are found using OpenAI embedding API and cosine simularity. Posts have between 0 and 15 related posts, depending on how many we've written in that area. 

Changes made:
- Table of contents defaults on
- TOC turns off in cases it doesn't look good or breaks the related item tie in
- Wide images turned off on posts where wide image would now break the layout of sidebar ( because its near the top)

Still to do:
- Related items is calculated in suggested_posts.py and written to related_articles.yml. Need some automation do this for new articles.

Posts vary in a number of way: Some have TOC and are in a series:

Examples:
With TOC and part of series:
https://66045cfe5c4d842fe219540d--ephemeral-narwhal-3ac3b2.netlify.app/blog/g++-makefile/

No TOC but series:
https://66045cfe5c4d842fe219540d--ephemeral-narwhal-3ac3b2.netlify.app/blog/thought-leaders/

TOC but no series:
https://66045cfe5c4d842fe219540d--ephemeral-narwhal-3ac3b2.netlify.app/blog/why-developer-platform/

No TOC or Series:
https://66045cfe5c4d842fe219540d--ephemeral-narwhal-3ac3b2.netlify.app/blog/on-star-history/

### CODE REVIEW

File: blog/_posts/2022-07-02-software-engineering-daily.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 title: "Earthly on Software Engineering Daily"
 categories:
   - News
+toc: false
```
Comment (adamgordonbell) [inline blog/_posts/2022-07-02-software-engineering-daily.md:5]: Disabling table of contents, where it is just one item, way way too many items, or somehow blows up page layout.

**Comment by thtmnisamnstr** [comment]
I'm slightly confused about the difference between the "Recommended for You", "Spotlight on Related Content", and "Discover More" sections. They seem kinda redundant.

Also, in the Series sections, the title always fits and is never truncated (or so it seems), but in all 3 of the previously mentioned sections, the title frequently truncates. It seems like we should be able to make them not truncate as well.

**Comment by adamgordonbell** [comment]
> I'm slightly confused about the difference between the "Recommended for You", "Spotlight on Related Content", and "Discover More" sections. They seem kinda redundant.

There is no difference, its 0-3 in teh first section, 4-8 in the next and 9 to 15 in the last. Just a way to break it up visually for the user. If you have better idea for naming it, let me know.

> Also, in the Series sections, the title always fits and is never truncated (or so it seems), but in all 3 of the previously mentioned sections, the title frequently truncates. It seems like we should be able to make them not truncate as well.

This is because when you make a series, you have to write a shorter version of the title to go with it and put it in navigation.yml. So we'd need a way to write 450 shorter titles to make that work.

**Comment by vladaionescu** [comment]
Great! I have similar questions as Gavin. Re truncation - at the very least we could make it show the full title as a tool tip on link hover? It would be nice if we didn't need truncation at all.

**Comment by thtmnisamnstr** [comment]
We could include shorter titles in the headmatter in each mdx file, but a way to generate 450 of them that are accurate is more difficult.

Instead of "Recommended for You" and "Spotlight on Related Content" on the same side and "Discover More" on the right side, how about we just include one on the left and one on the right. As far as naming for each section, I don't have any good ideas. Maybe "Related Content" and the left and "Discover More" on the right.

**Comment by vladaionescu** [comment]
Hmm.. I don't know if I would change the titles in the links necessarily. It might have negative SEO influence if the links don't match the title of the page.

**Pull request merged**